### PR TITLE
Fix misc bugs

### PR DESCRIPTION
**Pull request by ashwinvin**
# Description of Changes
FIxes the following:
- `env.set` and `remove` now actually sets and removes env in process level. This also fixes env vars not being updated by `export`
- prevent unstaged files from being committed. The pre-commit actually staged everything before linting which lead to this.
- remove use of non-existant api.

### CODE REVIEW

File: dev/hooks/pre-commit [hunk 1]
```diff
@@ -4,7 +4,7 @@ BOLD='\033[1;36m'
 NC='\033[0m'
 printf "${BOLD}[PRE-COMMIT]${NC} formatting code...\n"
 cargo +nightly fmt --all
-git add -u
```
Comment (MrPicklePinosaur) [inline dev/hooks/pre-commit:7]: The reason this is here is to restage files that were affected by cargo format, removing it has the chance that you forget to stage files that were formatted. Perhaps there is a way to find out which files were affected by the format and stage only those.

I would leave this in for now

**Review by MrPicklePinosaur — CHANGES_REQUESTED** [review]

File: dev/hooks/pre-commit (hunk 1 above)
Comment (ashwinvin) [reply dev/hooks/pre-commit:7]: I think switching to [pre-commit](https://pre-commit.com/) is the best choice here. Its quite easy to setup, extensible and automatically handles cases like this

Comment (MrPicklePinosaur) [reply dev/hooks/pre-commit:7]: I'll take a look at this. If possible I would like to avoid additional dependencies for developers however

**Review by MrPicklePinosaur — APPROVED** [review]
lgtm

**Pull request merged**